src/main/java/com/mrcrayfish/controllable/client/RenderEvents.java

```diff
@@ -35,6 +35,8 @@
 {
     private static final ResourceLocation CONTROLLER_BUTTONS = new ResourceLocation(Reference.MOD_ID, "textures/gui/buttons.png");
 
+    private float prevHealth = -1;
```

Comment:
This value needs to be reset back to -1 when the player exits the world, otherwise if they load another save and their health is lower it will cause the vibration to happen on world load. I have tested this. Besides that, all the code looks good. Any reason why you decided to put the test in a tick event rather than LivingDamageEvent?

Reply:
The LivingDamageEvent and LivingHurtEvent both only fire server side.

Reply:
Reset prevHealth when world is null;

Reply:
Makes sense. Changes look good, will merge now.